docs/en/reference/dql-doctrine-query-language.rst

```diff
@@ -608,6 +608,68 @@ You can also nest several DTO :
 	
 Note that you can only pass scalar expressions or other Data Transfer Objects to the constructor.
 
+If you use your data-transfer objects for multiple queries but not necessarily with the same parameters,
+you can use named arguments or variadic arguments, add ``named`` keyword in your DQL querie before your Dto :
+
+.. code-block:: php
+
+    <?php
+
+    class CustomerDTO
+    {
+        public function __construct(
+                public string|null $name = null,
+                public string|null $email = null,
+                public string|null $city = null,
+                public mixed|null $value = null,
+                public AddressDTO|null $address = null,
+        ) {
+        }
+    }
+
+And then you can select the fields you want in the order you want, and the ORM will try to match argument names with the selected field names :
+
+.. code-block:: php
+
+    <?php
+    $query = $em->createQuery('SELECT NEW NAMED CustomerDTO(a.city, c.name) FROM Customer c JOIN c.address a');
+    $users = $query->getResult(); // array of CustomerDTO
+
+    // CustomerDTO => {name : 'SMITH', email: null, city: 'London', value: null}
+
+ORM will also look column aliases before columns names :
+
+.. code-block:: php
+
+    <?php
+    $query = $em->createQuery('SELECT NEW NAMED CustomerDTO(c.name, CONCAT(a.city, ' ' , a.zip) AS value) FROM Customer c JOIN c.address a');
+    $users = $query->getResult(); // array of CustomerDTO
+
+	// CustomerDTO => {name : 'DOE', email: null, city: null, value: 'New York 10011'}
+
+For aliases a field you can use SQL notation with `AS` keyword, or the PHP named arguments notation :
+
+.. code-block:: php
+
+	<?php
+	$query = $em->createQuery('SELECT NEW NAMED CustomerDTO(name: c.name, value: CONCAT(a.city, ' ' , a.zip)) FROM Customer c JOIN c.address a');
+	$users = $query->getResult(); // array of CustomerDTO
+
+	// CustomerDTO => {name : 'DOE', email: null, city: null, value: 'New York 10011'}
+
+The `NAMED` keywork must be added to all dto you want to instantiate :
+
+.. code-block:: php
+
+	<?php
+	$query = $em->createQuery('SELECT NEW NAMED CustomerDTO(name: c.name, address: NEW NAMED AddressDTO(a.street, a.city, a.zip)) FROM Customer c JOIN c.address a');
+	$users = $query->getResult(); // array of CustomerDTO
+
+	// CustomerDTO => {name : 'DOE', email: null, city: null, value: 'New York 10011'}
+
+Be careful, if two fields have the same name/alias, second will erase first.
+If you put an argument name empty (for functions) or not in constructor argument list, ORM will throw an error.
```

Comment:
An error or an exception?

Reply:
actually an error, thorwn by PHP and ORM : "Error: Unknown named parameter $"

Reply:
Sorry, where in the code is it thrown? I can't find it.